crates/reth/evm/src/precompiles/bridge.rs

```diff
@@ -33,13 +38,13 @@ impl BridgeoutPrecompile {
 impl<DB: Database> ContextStatefulPrecompile<DB> for BridgeoutPrecompile {
     fn call(
         &self,
-        dest_pk_bytes: &Bytes,
+        destination_bytes: &Bytes,
         _gas_limit: u64,
         evmctx: &mut revm::InnerEvmContext<DB>,
     ) -> PrecompileResult {
-        // Validate the length of the destination public key
-        let dest_pk = try_into_pubkey(dest_pk_bytes)
-            .map_err(|_| PrecompileError::other("Invalid public key length: expected 32 bytes"))?;
+        // Validate that this is a valid BOSD
+        let destination = try_into_bosd(destination_bytes)?;
+        let destination_bytes = destination.to_bytes().into();
```

Comment:
Does it make sense to reencode it to bytes like this?  Or could we just copy the calldata bytes after we've validated it?  The current feels safer but it feels like maybe it does an extra copy we wouldn't normally have to do.